### PR TITLE
Optimize CodeQL workflow

### PR DESCRIPTION
**Pull request by armandgrillet**
**What this PR does / why we need it**:

Instead of having one CodeQL workflow running go, javascript, and python checks for each PR modifying go, javascript, or python files, we now have specialized workflows so that only the right workflow runs depending on the files modified in a PR.

The main workflow has also been optimized to not run when a recent commits only edited text or configuration files.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: .github/workflows/pr-codeql-analysis-go.yml
Comment (marefr) [inline .github/workflows/pr-codeql-analysis-go.yml:17]: Nit. When running each language on it's own might not need to use the matrix feature? Do we want to fail fast in PR's? Maybe we don't

**Review by marefr — APPROVED** [review]
This seems good I think. Added a comment/question.

**Review by marefr — APPROVED** [review]
LGTM, please follow up after a day when this has been merged to make sure it works as expected :smile:

File: .github/workflows/pr-codeql-analysis-javascript.yml
Comment (leeoniya) [inline .github/workflows/pr-codeql-analysis-javascript.yml:8]: `*.tsx`, too

**Review by leeoniya — APPROVED** [review]

**Pull request merged**
